api/app.py

```diff
@@ -42,25 +49,65 @@
 PalaceXrayProfiler.configure(app)
 
 
-@app.before_first_request
+def initialize_application() -> Flask:
+    with app.app_context(), flask_babel.force_locale("en"):
+        initialize_database()
+        initialize_circulation_manager()
+        initialize_admin()
+    return app
+
+
 def initialize_database(autoinitialize=True):
     testing = "TESTING" in os.environ
-
     db_url = Configuration.database_url()
     if autoinitialize:
         SessionManager.initialize(db_url)
     session_factory = SessionManager.sessionmaker(db_url)
     _db = flask_scoped_session(session_factory, app)
     app._db = _db
-
     log_level = LogConfiguration.initialize(_db, testing=testing)
     debug = log_level == "DEBUG"
     app.config["DEBUG"] = debug
     app.debug = debug
     _db.commit()
+
     logging.getLogger().info("Application debug mode==%r" % app.debug)
 
 
+def initialize_admin(_db=None):
+    if getattr(app, "manager", None) is not None:
+        setup_admin_controllers(app.manager)
+    _db = _db or app._db
+    _db.execute("LOCK TABLE configurationsettings IN ACCESS EXCLUSIVE MODE;")
```

Comment:
This lock is required. Otherwise multiple processes initializing the app for the first time end up trying to write configuration to the database at the same time.